src/serac/physics/thermal_conduction_functional.hpp

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file thermal_conduction.hpp
+ *
+ * @brief An object containing the solver for a thermal conduction PDE
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+
+#include "serac/physics/base_physics.hpp"
+#include "serac/numerics/odes.hpp"
+#include "serac/numerics/stdfunction_operator.hpp"
+#include "serac/numerics/functional/functional.hpp"
+#include "serac/physics/state/state_manager.hpp"
+
+namespace serac {
+
+struct ConstantIsotropicMaterial {
+  double rho;
+  double cp;
+  double kappa;
+
+  template <typename T1, typename T2>
+  SERAC_HOST_DEVICE T2 thermalFlux([[maybe_unused]] T1& u, [[maybe_unused]] T2& du_dx) const
+  {
+    return kappa * du_dx;
+  }
+};
+
+struct ScalarFunction {
+  double constant;
+
+  template <typename T1>
+  SERAC_HOST_DEVICE double operator()([[maybe_unused]] T1& x, [[maybe_unused]] double t) const
+  {
+    return constant;
+  }
+};
+
+/**
+ * @brief An object containing the solver for a thermal conduction PDE
+ *
+ * This is a generic linear thermal diffusion oeprator of the form
+ *
+ *    M du/dt = -kappa Ku + f
+ *
+ *  where M is a mass matrix, K is a stiffness matrix, and f is a
+ *  thermal load vector.
+ */
+template <int order, int dim>
+class ThermalConductionFunctional : public BasePhysics {
+public:
+  /**
+   * @brief A timestep method and config for the M solver
+   */
+  struct TimesteppingOptions {
+    /**
+     * @brief The timestepping method to be applied
+     *
+     */
+    TimestepMethod timestepper;
+
+    /**
+     * @brief The essential boundary enforcement method to use
+     *
+     */
+    DirichletEnforcementMethod enforcement_method;
```

Comment:
minor: could we consider adopting a convention where we use a single-line or same-line doxygen comments for short descriptions like this? e.g. 
```suggestion
  /// @brief A timestep method and config for the M solver
  struct TimesteppingOptions {
  
    /// @brief The timestepping method to be applied
    TimestepMethod timestepper;

    /// @brief The essential boundary enforcement method to use
    DirichletEnforcementMethod enforcement_method;
```

I think its valuable to be able to see more of a sourcefile at once, and these multi-line doxygen expressions for short messages occupy a lot of vertical space in an editor.

Reply:
I agree. I admit I default to this because my IDE autofills a lot of this.